configure.ac

```diff
@@ -0,0 +1,307 @@
+	if test "x$enable_single" = "xyes"; then
+		AC_SUBST(FFTW3LDADD, -lfftw3f)
+	else
+		AC_SUBST(FFTW3LDADD, -lfftw3)
+	fi
+else
+if test "x$with_mkl" != "xno" -a "x$with_mkl" != "xcheck"; then
+	AC_MSG_NOTICE([FFT engine mkl was provided])
+	AC_SUBST(FFTENGINE, mkl)
+	AC_SUBST(MKLFCFLAGS, ${MKL_FCFLAGS})
+        AC_SUBST(MKLLDFLAGS, ${MKL_LDFLAGS})
+else
+	AC_MSG_NOTICE([FFT engine is generic])
+	AC_SUBST(FFTENGINE, generic)
+fi
+fi
+AM_CONDITIONAL(HAVE_FFTW3, [test "x$with_fftw3" != "xno" -a "x$with_fftw3" != "xcheck"])
+AM_CONDITIONAL(HAVE_MKL, [test "x$with_mkl" != "xno" -a "x$with_mkl" != "xcheck"])
+
+#
+# pfunit unit tests
+#
+AC_ARG_WITH(pfunit,
+            [AS_HELP_STRING([--with-pfunit=PATH],[prefix directory for pfunit])],
+            [if test "x$withval" = "x"; then
+                with_pfunit=no
+            fi],
+            [with_pfunit=no])
+if test "x$with_pfunit" != "xno"; then
+        AC_MSG_NOTICE([Unit tests activated])
+        AC_SUBST(PFUNITDIR, ${with_pfunit})
+fi
+AM_CONDITIONAL(HAVE_PFUNIT, [test "x$with_pfunit" != "xno"])
+
+#
+# Check the Fortran compiler
+#
+Fcompiler=none
+#
+# 1. gfortran ?
+#
+if test "x$Fcompiler" != "xok"; then
+echo `$FC --version 2>&1 | head -1` | grep 'GNU Fortran' > /dev/null
+if test "$?" = "0" ; then
+	Fcompiler=ok
+	fc_version="`$FC -dumpversion | cut -f1 -d.`"
+	AC_MSG_NOTICE([GNU Fortran version $fc_version])
+	if test "x$enable_single" = "xyes"; then
+		FCFLAGS="-cpp"
+	else
+                FCFLAGS="-cpp -DDOUBLE_PREC"
+		if test "x$enable_single_snapshot" = "xyes"; then
+			FCFLAGS="$FCFLAGS -DSAVE_SINGLE"
+		fi
+        fi
+	AS_IF([test "x$enable_debug" = "xyes"],
+	      [FCFLAGS="$FCFLAGS -Og -g3 -ffpe-trap=invalid,zero -fcheck=all -fimplicit-none -DDEBUG"],
+	      [FCFLAGS="$FCFLAGS -O3 -DNDEBUG"])
+	OMPFLAG="-fopenmp"
+fi
+fi
+#
+# 2. ifort classic ?
+#
+if test "x$Fcompiler" != "xok"; then
+echo `$FC --version 2>&1 | head -1` | grep 'ifort' > /dev/null
+if test "$?" = "0"; then
+	Fcompiler=ok
+	fc_version="`$FC -dumpversion | cut -f1 -d.`"
+	AC_MSG_NOTICE([Intel ifort version $fc_version])
+        if test "x$enable_single" = "xyes"; then
+                FCFLAGS="-fpp"
+        else
+                FCFLAGS="-fpp -DDOUBLE_PREC"
+                if test "x$enable_single_snapshot" = "xyes"; then
+                        FCFLAGS="$FCFLAGS -DSAVE_SINGLE"
+                fi
+        fi
+	AS_IF([test "x$enable_debug" = "xyes"],
+	      [FCFLAGS="$CFLAGS -g -O0 -traceback -check all -check nopointer -fpe0 -ftrapuv -DDEBUG"],
+	      [FCFLAGS="$FCFLAGS -O3 -DNDEBUG"])
+	OMPFLAG="-qopenmp"
+fi
+fi
+#
+# 3. nvfortran ?
+#
+
+#
+# 4. TODO (Cray, IBM XL, ...)
+#
+
+#
+# Provide the git version
+# or the one defined in AC_INIT
+#
+if test -d ${srcdir}/.git; then
+	X3DVERSION="`git $(srcdir) describe --tag --long --always`"
+	# Add "-modif" if any diff
+	echo `git diff $(srcdir)` | grep index > /dev/null
+	if test "$?" = "0"; then
+		X3DVERSION="${X3DVERSION}-modif"
```

Comment:
@pbartholomew08 @rfj82982 this could be a nice addition : provide the commit in the listing and add a marker if there is a diff.
We could even save the diff between the commit and the current code somewhere so that the code can be reproduced exactly.

Reply:
I agree, though I would fear the outcome if the diff becomes significant. I was actually thinking about something related to this - it would be a nice feature if we could generate some machine readable (e.g. json, but open to suggestions) metadata file with a run, e.g.

- compiler and flags (already there)
- problem size
- number of mpi ranks
- number of threads
- etc.

A lot of this is already output by the code, but it would be nice if it were *easily* parse-able to automate analysing scaling results, etc.

Reply:
This kind of things could be extracted from the listing using [Py4Incompact3D](https://github.com/xcompact3d/Py4Incompact3D) ?